lib/reports/BackupReportManager.js

```diff
@@ -57,18 +55,24 @@ class BackupReportManager {
           backupSuccessCount += (endIndex - startIndex);
           if (startIndex === endIndex) {
             noBackupDays.push(currentDay);
-            if (
-              (secondlastFailDay && lastFailDay) &&
-              (secondlastFailDay.valueOf() === moment(lastFailDay).subtract(1, 'days').toDate().valueOf()) &&
-              (lastFailDay.valueOf() === moment(currentDay).subtract(1, 'days').toDate().valueOf())
-            ) {
-              threeConsecutiveBackupFailureCount++;
-            }
-            secondlastFailDay = lastFailDay;
-            lastFailDay = currentDay;
           }
           startIndex = endIndex;
         }
+        // check max_allowed_no_backup_days_in_row failure count
+        const maxAllowedNoBackupDaysInRow = config.reporting.max_allowed_no_backup_days_in_row;
+        let windowStartIndex = 0; // start index of consecutive no backup days window
+        let windowEndIndex = 0; // end index of consecutive no backup days window
+        while (windowEndIndex < noBackupDays.length) {
+          while (
+            windowEndIndex + 1 < noBackupDays.length &&
+            noBackupDays[windowEndIndex].valueOf() === moment(noBackupDays[windowEndIndex + 1]).subtract(1, 'days').toDate().valueOf()
+          ) {
+            windowEndIndex++;
+          }
+          threeConsecutiveBackupFailureCount += (windowEndIndex - windowStartIndex + 1) >= maxAllowedNoBackupDaysInRow ? (windowEndIndex - windowStartIndex + 2 - maxAllowedNoBackupDaysInRow) : 0;
```

Comment:
can you rename these variables now 

- `threeConsecutiveBackupFailureCount` to `consecutiveBackupFailureCount`
- `failedBackups3daysInARow` to `failedBackupsNdaysInARow` ?